server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -111,9 +111,6 @@ public InternalClusterInfoService(Settings settings, ClusterService clusterServi
         clusterSettings.addSettingsUpdateConsumer(INTERNAL_CLUSTER_INFO_UPDATE_INTERVAL_SETTING, this::setUpdateFrequency);
         clusterSettings.addSettingsUpdateConsumer(DiskThresholdSettings.CLUSTER_ROUTING_ALLOCATION_DISK_THRESHOLD_ENABLED_SETTING,
                                                   this::setEnabled);
-
-        // listen for state changes (this node starts/stops being the elected master, or new nodes are added)
-        clusterService.addListener(this);
```

Comment:
Sorry for the conflicts with #59880, but I think we need this line still.

Reply:
Oh wait, no I see this got moved to `Node`. That's nicer indeed but I expect `InternalClusterInfoServiceSchedulingTests` need adjusting too.

Reply:
No need to apologize. Addressed in https://github.com/elastic/elasticsearch/pull/59932/commits/59a7feee7cc2ec9ce13f353a3af798e6eaa06116